airbyte-commons/src/main/java/io/airbyte/commons/util/ResourceIterator.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.commons.util;
+
+import java.util.Iterator;
+
+/**
+ * If you operate on this iterator, you better close it. {@link ResourceIterator#close} must be
+ * idempotent. The contract on this interface is that it may be called MANY times.
+ *
+ * @param <T> type
+ */
+public interface ResourceIterator<T> extends Iterator<T>, AutoCloseable {}
```

Comment:
would a better name be `AutoCloseableIterator`?

Reply:
I'm trying to invoke the `Resource` concept from `try-with-resource`. This was originally called `CloseableIterator` which felt to me like it undersold what was going on a little bit.

Reply:
Agreed that Closeable isn't the best but AutoCloseable already represent the concept of try-with-resource for a java programmer. So we should use the same convention

Reply:
kk

Reply:
done.